app/main.py

```diff
@@ -1,2 +1,102 @@
+from typing import Any
+
+
+class Node:
+    def __init__(self, key: Any, value: Any, next_node: Any = None) -> None:
+        self.key = key
+        self.value = value
+        self.next = next_node
+
+
 class Dictionary:
-    pass
+    def __init__(
+            self, initial_capacity: int = 8,
+            load_factor: float = 2 / 3
+    ) -> None:
+        self.capacity = initial_capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.table = [None] * self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        index = self._hash_function(key) % self.capacity
+        node = self.table[index]
+
+        while node:
```

Comment:
You can put index lookup in a separate method and reuse it in this and other methods to keep it DRY